### ISSUE TITLE
Error: There is no document formatter for 'terraform'-files installed.

### ISSUE THREAD

**Issue by dominik-fielmann** [hashicorp/vscode-terraform#270]
I can't get the terraform formatter to work.

I am running on Ubuntu 19.10 with snap packages of vscode and terraform.

When I format a .tf file, it says:
>  There is no document formatter for 'terraform'-files installed.

I tried multiple settings, nothing worked:
```
    "editor.defaultFormatter": "esbenp.prettier-vscode",
    "[terraform]": {
      "editor.defaultFormatter": "mauve.terraform"
    },
    "editor.formatOnSave": true,
    "terraform.path": "/snap/bin/terraform"
``` 

Reinstalling the plugin did not fix it either.

Any ideas?

---

<details>
  <summary>code --version</summary>

1.41.1
26076a4de974ead31f97692a0d32f90d735645c0
x64
</details>

<details>
  <summary>code --list-extensions --show-versions</summary>

dbaeumer.vscode-eslint@2.0.14
donjayamanne.githistory@0.4.13
eamodio.gitlens@10.2.0
esbenp.prettier-vscode@3.18.0
fwcd.kotlin@0.2.11
GrapeCity.gc-excelviewer@2.1.32
jock.svg@1.1.0
mauve.terraform@1.4.0
ms-azuretools.vscode-docker@0.9.0
ms-kubernetes-tools.vscode-kubernetes-tools@1.0.9
ms-vsliveshare.vsliveshare@1.0.1482
Orta.vscode-jest@3.1.1
pflannery.vscode-versionlens@0.24.0
redhat.vscode-yaml@0.6.1
shd101wyy.markdown-preview-enhanced@0.5.1
thibaudcolas.stylelint@0.51.0
</details>

**Comment by tracphil** [comment]
For `[terraform]`, take what you have out and put this instead and see how it works:

```json
    "[terraform]": {
        "editor.formatOnSave": true
    },
```

**Comment by dominik-fielmann** [comment]
> For `[terraform]`, take what you have out and put this instead and see how it works:

How would that help? I already have `formatOnSave` enabled not just for terraform but for everything.

I just tried it, doesn't work. 

formatOnSave can only work if there is a formatter assigned and there is none (see the error message I provided). So nothing happens on save, and if I try to format with the `Format document` command, it fails with the error messsage from above.

Somehow it does not recognize the formatter for terraform even though the plugin is installed.

**Comment by tracphil** [comment]
Dunno then. This is my complete config for terraform:

```json
{
  "[terraform]": {
    "editor.formatOnSave": true
  },
  "editor.defaultFormatter": "esbenp.prettier-vscode",
  "terraform.codelens.enabled": true,
  "terraform.format": {
      "enabled": true,
      "formatOnSave": true,
      "ignoreExtensionsOnSave": [
        ""
      ]
    },
  "terraform.indexing": {
      "enabled": false,
      "liveIndexing": false,
      "delay": 500,
      "exclude": [
          ".terraform/**/*",
          "**/.terraform/**/*"
      ]
  },
  "terraform.languageServer": {
      "enabled": true,
      "installCommonProviders": true,
      "pathToBinary": "/usr/local/bin/",
      "args": []
  },
  "terraform.lintPath": "/usr/local/bin/tflint"
}
```

**Comment by dominik-fielmann** [comment]
My suspicion is that this could be a bug related to snap packages. Any help debugging the problem would be appreciated. The output of the "Terraform" log window is completely blank when this happens.

**Comment by paultyng** [comment]
We just released [v2.0.0-rc.1](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.0.0-rc.1) of the extension. The main features include:

* Added syntax support for 0.12
* Added [terraform-ls](https://github.com/hashicorp/terraform-ls) usage by default (currently on 0.3.0, which offers basic provider code completion)

You can find additional information and specifics in the [release notes](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.0.0-rc.1) and [CHANGELOG](https://github.com/hashicorp/vscode-terraform/blob/master/CHANGELOG.md).

With this release we expect that many of the prior issues and PRs are no longer relevant or have been addressed, and are therefore being closed. If you feel the action taken on an issue or PR is in error, please comment as such and we can figure out the appropriate way to address it.

We plan to add the final 2.0.0 release to the marketplace soon, but are actively seeking your feedback now on the release candidates. You can download the .vsix from the releases page and [manually install it in VS Code](https://code.visualstudio.com/docs/editor/extension-gallery#_install-from-a-vsix) to try it out.
